### PR TITLE
Validate result

### LINKED ISSUE #7966: Validate results after SingleTestRun/EnsembleExperiment
Sometimes it is a bit hard for users to tell if all the required observations and parameters are loaded in the callback. In AHM we allow some observations to be missing, and only crash if all observations are deactivated.

The REFCASE has some functionality to check that the correct dates are present for summary data, however there has historically been a problem that it is not necessarily in sync with the schedule file, and as such, might give false positives or negatives.

One way we can improve this, is by validating that SingleTestRun and EnsembleExperiment validate the results for missing parameters and responses. This can be as simple as raising an exception when it has completed if something is missing, and everything should show up correctly in the gui.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
Can you elaborate in the git commit message what is validated?

**Review by oyvindeide — COMMENTED** [review]
Sorry, forgot to publish my review

File: tests/unit_tests/run_models/test_ensemble_experiment.py [hunk 1]
```diff
@@ -45,3 +49,92 @@ def get_run_path_mock(realizations, iteration=None):
...
+
+        EnsembleExperiment.validate = MagicMock()
+        ensemble_experiment = EnsembleExperiment(
+            simulation_arguments, MagicMock(), None, None, MagicMock()
+        )
+
+        ensemble_experiment._validate_results(prior_context)
+
+
+def test_validate_results_with_error(snake_oil_default_storage):
```
[the remark this answers is not in the document]
Comment (frode-aarstad) [reply tests/unit_tests/run_models/test_ensemble_experiment.py:88]: > Tested manually by modifying the snake oil case, removed the refcase and timemap, removed the `FOPR` observation and the HISTORY_SOURCE keyword as well as change `DATE`-> `RESTART` for `WPR_DIFF`, then added a new `SUMMARY_OBSERVATION` with a much newer date.
> 
> So in the context of this test, that is more like line 99, except that the name is `FOPR` instead, and the date does not exist in the responses.
> 
> What is already here also makes sense, as it simulates a missing key in the response file.

I will try to make a test doing what you describe here

**Comment by oyvindeide** [comment]
Think this looks good, but needs a rebase as there has been some changes to the run models, should be minor conflicts though.

File: src/ert/cli/main.py [hunk 2]
```diff
@@ -145,5 +145,4 @@ def run_cli(args: Namespace, plugin_manager: Optional[ErtPluginManager] = None)
 
     if end_event.failed:
         # If monitor has not reported, give some info if the job failed
-        msg = end_event.msg if args.disable_monitoring else ""
-        raise ErtCliError(msg)
```
Comment (frode-aarstad) [inline src/ert/cli/main.py:149]: @yngve-sk / @oyvindeide  Is there a good reason for not propagating the error message here if monitoring is enabled?

Comment (oyvindeide) [reply src/ert/cli/main.py:149]: The monitor has already reported the error message, so it came out double

**Comment by oyvindeide** [comment]
What is the status of this PR?

**Comment by frode-aarstad** [comment]
Closing this due to stale code. Probably easier to reimplement with new eyes on the problem

**Pull request closed without merging**